geaflow/geaflow-dsl/geaflow-dsl-connector/geaflow-dsl-connector-jdbc/src/main/java/com/antgroup/geaflow/dsl/connector/jdbc/JDBCTableSource.java

```diff
@@ -0,0 +1,162 @@
+    private String username;
+    private String password;
+    private String tableName;
+    private Connection connection;
+    private Statement statement;
+
+    @Override
+    public void init(Configuration tableConf, TableSchema tableSchema) {
+        LOGGER.info("prepare with config: {}, \n schema: {}", tableConf, tableSchema);
+        this.tableConf = tableConf;
+        this.schema = tableSchema;
+
+        this.driver = tableConf.getString(JDBCConfigKeys.GEAFLOW_DSL_JDBC_DRIVER);
+        this.url = tableConf.getString(JDBCConfigKeys.GEAFLOW_DSL_JDBC_URL);
+        this.username = tableConf.getString(JDBCConfigKeys.GEAFLOW_DSL_JDBC_USERNAME);
+        this.password = tableConf.getString(JDBCConfigKeys.GEAFLOW_DSL_JDBC_PASSWORD);
+        this.tableName = tableConf.getString(JDBCConfigKeys.GEAFLOW_DSL_JDBC_TABLE_NAME);
+    }
+
+    @Override
+    public void open(RuntimeContext context) {
+        try {
+            Class.forName(this.driver);
+            this.connection = DriverManager.getConnection(url, username, password);
+            this.statement = connection.createStatement();
+        } catch (Exception e) {
+            throw new GeaFlowDSLException("failed to connect to database", e);
+        }
+    }
+
+    @Override
+    public List<Partition> listPartitions() {
+        return Collections.singletonList(new JDBCPartition(this.tableName));
+    }
+
+    @Override
+    public <IN> TableDeserializer<IN> getDeserializer(Configuration conf) {
+        return (TableDeserializer<IN>) new RowTableDeserializer();
+    }
+
+    @Override
+    public <T> FetchData<T> fetch(Partition partition, Optional<Offset> startOffset,
+                                  long windowSize) throws IOException {
+        if (!partition.getName().equals(this.tableName)) {
+            throw new GeaFlowDSLException("wrong partition");
+        }
+        long offset = 0;
+        if (startOffset.isPresent()) {
+            offset = startOffset.get().getOffset();
+        }
+        List<Row> dataList;
+        try {
+            dataList = JDBCUtils.selectRowsFromTable(this.statement, this.tableName,
+                this.schema.size(), offset, windowSize);
+        } catch (SQLException e) {
+            throw new GeaFlowDSLException("select rows form table failed.", e);
+        }
+        JDBCOffset nextOffset = new JDBCOffset(offset + windowSize);
+        return (FetchData<T>) new FetchData<>(dataList, nextOffset, false);
+    }
+
+    @Override
+    public void close() {
+        try {
+            if (statement != null) {
+                statement.close();
+                statement = null;
+            }
+            if (connection != null) {
+                connection.close();
+                connection = null;
+            }
+        } catch (SQLException e) {
+            throw new GeaFlowDSLException("failed to close connection.");
+        }
+    }
+
+    public static class JDBCPartition implements Partition {
```

Comment:
Subclasses of Partition must implement the hashCode and equals methods, as GeaFlow will use the equals comparison to determine if the source partition has changed, and subsequently, restart and reopen the source.